### PR TITLE
Fixes prefilled drinking glasses, and small carton icon

### LINKED ISSUE #72468: Prefilled drinkinglass bottles look like regular drinking glasses, and runtime on update appearance
## Round ID:

N/A

## Testmerges:

N/A

## Reproduction:

Get a bottle of /obj/item/reagent_containers/cup/glass/drinkingglass/filled/nuka_cola (or cola or soda), see that it looks like a generic black liquid in the glass. Drink from it, notice runtime.

If you pour it into a regular cup, it will work perfectly, as /obj/item/reagent_containers/cup/glass/drinkingglass is registered in GLOB.glass_style_singletons

### PR DESCRIPTION
**Pull request by Tastyfish**
## About The Pull Request

Fixes #72468

Fixes a few drink container issues:
- The `/obj/item/reagent_containers/cup/glass/drinkingglass/filled` subtypes runtimed whenever their reagents changed because of the strict way that `/datum/component/takes_reagent_appearance` compares container types.
  - `/datum/component/takes_reagent_appearance` now allows for an alternate type to check against `glass_style.required_container_type` via a var called `base_container_type`. The filled glasses now set this var to the main drinking glass type.
- As well, filled glasses didn't have their appearance set up to match the corresponding glass style. Thus, the `/obj/item/reagent_containers/cup/glass/drinkingglass/filled` type now updates its appearance on initialization.

- Seperately, the small carton's appearance broke if you put a reagent in that doesn't match a glass style, reverting to the "water_cup" icon state which doesn't exist in the boxes.dmi file. This is because it was a subtype of sillycup, but there is nothing gained as far as I can see from that type relationship, so the small carton was repathed to `/obj/item/reagent_containers/cup/glass/smallcarton`.

### Tested all 3 users of takes_reagent_appearance:
image
## Why It's Good For The Game

Runtime fix, fixed visuals.
## Changelog
:cl:
fix: The small carton no longer randomly disappears.
fix: Pre-filled bottles of cola & nuka cola, and glasses of soda now have the correct sprite.
/:cl:

### CODE REVIEW

File: code/modules/reagents/reagent_containers/cups/drinks.dm [hunk 1]
```diff
@@ -366,27 +366,29 @@
 	icon_state = reagents.total_volume ? "water_cup" : "water_cup_e"
 	return ..()
 
-/obj/item/reagent_containers/cup/glass/sillycup/smallcarton
+/obj/item/reagent_containers/cup/glass/smallcarton
```
Comment (JohnFulpWillard) [inline code/modules/reagents/reagent_containers/cups/drinks.dm:369]: Why not subtype of ``/obj/item/reagent_containers/cup/glass/bottle/juice``

Comment (Tastyfish) [reply code/modules/reagents/reagent_containers/cups/drinks.dm:369]: It looks like the only thing gained from doing that is letting a person kill themselves with a cardboard molotov. Which is funny, so sure. One moment.

Comment (Tastyfish) [reply code/modules/reagents/reagent_containers/cups/drinks.dm]: @JohnFulpWillard Done.

File: code/modules/reagents/reagent_containers/cups/drinkingglass.dm [hunk 2]
```diff
@@ -71,6 +80,13 @@
 	else
 		desc = "A shot glass - the universal symbol for bad decisions."
 
+/obj/item/reagent_containers/cup/glass/drinkingglass/filled
+	base_container_type = /obj/item/reagent_containers/cup/glass/drinkingglass
+
+/obj/item/reagent_containers/cup/glass/drinkingglass/filled/Initialize(mapload, vol)
+	. = ..()
+	update_appearance()
```
Comment (JohnFulpWillard) [inline code/modules/reagents/reagent_containers/cups/drinkingglass.dm:88]:
```suggestion
	update_appearance(UPDATE_ICON)
```

File: code/modules/reagents/reagent_containers/cups/drinkingglass.dm [hunk 3]
```diff
@@ -71,6 +80,13 @@
 	else
 		desc = "A shot glass - the universal symbol for bad decisions."
 
+/obj/item/reagent_containers/cup/glass/drinkingglass/filled
+	base_container_type = /obj/item/reagent_containers/cup/glass/drinkingglass
+
+/obj/item/reagent_containers/cup/glass/drinkingglass/filled/Initialize(mapload, vol)
+	. = ..()
+	update_appearance(UPDATE_ICON)
```
Comment (JohnFulpWillard) [inline code/modules/reagents/reagent_containers/cups/drinkingglass.dm:88]:
```suggestion
	update_appearance()
```
actually i was wrong

**Pull request merged**